docs/best_practices.md

```diff
@@ -0,0 +1,94 @@
+
+### Leverage TypeScript to prevent runtime bugs
+
+We use [TypeScript](https://www.typescriptlang.org/docs) to maximize runtime code safety.
+
+### Use Relay for network requests
+
+Data should be loaded from [Metaphysics](https://github.com/artsy/metaphysics), Artsy's GraphQL server. Requests to Metaphysics should be made through [Relay](https://relay.dev).
+
+- [Why Artsy uses Relay](http://artsy.github.io/blog/2017/02/05/Front-end-JavaScript-at-Artsy-2017/#Relay)
+- [Artsy JavaScriptures seminar on Relay](https://github.com/artsy/javascriptures/tree/master/4_intro-to-relay)
+- Examples
+  - [A top-level route-based Relay request](https://github.com/artsy/force/blob/0b291f005763e7c2600a5077786c9510bf655079/src/v2/Apps/Consign/consignRoutes.tsx#L28-L34)
+  - [A fragment container](https://github.com/artsy/force/blob/0b291f005763e7c2600a5077786c9510bf655079/src/v2/Apps/Consign/Routes/Offer/OfferDetailApp.tsx#L47-L57)
+
+### Prefer Relay containers (higher order components) over hooks
+
+We have a preference for Relay containers due to [`relay-hooks`](https://github.com/relay-tools/relay-hooks) hooks not being compatible with Relay containers which represent the majority of our components using Relay. (This could change once Relay releases its official hooks implementation.)
+
+### Keep file structure organized
+
+Generally speaking, all new product-centric code is written [inside the `src/v2`](https://github.com/artsy/force/tree/1842553ad34475bc3b804f00c6410d7f23d64f65/src/v2) folder.
+
+Sub-apps are written [inside of `src/v2/Apps` folder](https://github.com/artsy/force/tree/1842553ad34475bc3b804f00c6410d7f23d64f65/src/v2/Apps).
+
+Within a sub-app, things are typically structured like so:
+
+```
+├── routes.tsx
+├── Components
+│   ├── __tests__
+│   │   └── Foo.jest.tsx
+│   │   └── Bar.jest.tsx
+│   ├── Foo.tsx
+│   ├── Bar.tsx
+├── Routes
+│   ├── Home
+│   │   ├── HomeApp.tsx
+│   │   ├── Components
+│   │   │   ├── Foo.tsx
+│   │   │   ├── Bar.tsx
+│   │   ├── Utils
+│   │   │   └── formatCentsToDollars.ts
+│   └── Offer
+│       ├── OfferDetailApp.tsx
+│       ├── Components
+│       │   ├── Foo.tsx
+│       │   ├── Bar.tsx
+```
+
+If there's a component that might be shared across multiple sub-apps, it should go in the top-level [`/v2/Components`](https://github.com/artsy/force/tree/1842553ad34475bc3b804f00c6410d7f23d64f65/src/v2/Components) folder.
+
+Framework code is located in [`v2/Artsy`](https://github.com/artsy/force/tree/1842553ad34475bc3b804f00c6410d7f23d64f65/src/v2/Artsy).
+
+### Write unit tests for new components
+
+We use [Jest](https://jestjs.io/) for our unit tests, and [Mocha](https://mochajs.org/) for testing legacy code.
+
+Some top-level notes:
+
+- We use [`enzyme`](https://enzymejs.github.io/enzyme/)
+- We don't like snapshot tests; they produce too much churn for too little value.
```

Comment:
Non-blocking: use a verb that is less subjective. I also don't like snapshot tests, but I don't think it is a universal opinion. There is no question that we *avoid* them though. Removing the subjectivity of this sentence should make it less likely for people to want to rewrite it.

```suggestion
- We avoid snapshot tests; they produce too much churn for too little value.
```

Reply:
ugh @pepopowitz - i thought i hit commit suggestion here, updating

Reply:
Updated here: https://github.com/artsy/force/commit/77c6c5c3512a981e2ca183a5ba2a2b2e4a5e2987